src/main/java/com/amazonaws/services/kinesis/leases/impl/LeaseCleanupManager.java

```diff
@@ -189,62 +185,72 @@ public LeaseCleanupResult cleanupLease(LeasePendingDeletion leasePendingDeletion
                     Set<String> childShardKeys = leaseFromDDB.getChildShardIds();
                     if (CollectionUtils.isNullOrEmpty(childShardKeys)) {
                         try {
+                            // throws ResourceNotFoundException
                             childShardKeys = getChildShardsFromService(shardInfo);
-
                             if (CollectionUtils.isNullOrEmpty(childShardKeys)) {
                                 LOG.error("No child shards returned from service for shard " + shardInfo.getShardId());
+                                // If no children shard is found in DDB and from service, then do not delete the lease
```

Comment:
I don't think this offers any safety since this is just retried either way on the next deletion run, are we adding this for unit testing?

Reply:
This will prevent the the lease deletion and log error as this closed and valid shard could not retrieve children info from anywhere. This is a final guard to protect against bad server response.